Src/FluentAssertions/Specialized/TaskCompletionSourceAssertions.cs

```diff
@@ -0,0 +1,105 @@
+        /// Asserts that the <see cref="Task"/> of the current <see cref="TaskCompletionSource{T}"/> will complete within the specified time.
+        /// </summary>
+        /// <param name="timeSpan">The allowed time span for the operation.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public Task<AndWhichConstraint<TaskCompletionSourceAssertions<T>, T>> CompleteWithinAsync(
+            TimeSpan timeSpan, string because = "", params object[] becauseArgs)
+        {
+            return this.AssertAsync(timeSpan, because, becauseArgs,
+                "Expected {context} to complete within {0}{reason}, but found <null>.",
+                "Expected {context:task} to complete within {0}{reason}.",
+                new object[] { timeSpan });
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="Task"/> of the current <see cref="TaskCompletionSource{T}"/> will complete within the specified time.
+        /// </summary>
+        /// <param name="timeSpan">The allowed time span for the operation.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public Task<AndWhichConstraint<TaskCompletionSourceAssertions<T>, T>> CompleteAsync(
+            string because = "", params object[] becauseArgs)
+        {
+            var timeSpan = AssertionOptions.TaskTimeout;
+            return this.AssertAsync(timeSpan, because, becauseArgs,
+                "Expected {context} to complete{reason}, but found <null>.",
+                "Expected {context:task} to complete{reason}.",
+                new object[0]);
+        }
+
+        private async Task<AndWhichConstraint<TaskCompletionSourceAssertions<T>, T>> AssertAsync(
+            TimeSpan timeSpan,
+            string because,
+            object[] becauseArgs,
+            string failMessage1,
+            string failMessage2,
+            object[] failArgs)
+        {
+            Execute.Assertion
+                .ForCondition(subject is object)
+                .BecauseOf(because, becauseArgs)
+                .FailWith(failMessage1, failArgs);
+
+            using (var timeoutCancellationTokenSource = new CancellationTokenSource())
```

Comment:
Maybe we can bump the version of C# to 8 so we don't need those blocks anymore

Reply:
Should I really change the project file this way?

Reply:
It's up to you. It was just a suggestion.